sdk/storage/Azure.Storage.Blobs/src/BlobClient.cs

```diff
@@ -449,21 +429,185 @@ public virtual async Task<Response<BlobContentInfo>> UploadAsync(string path)
             string path,
-            CancellationToken cancellationToken)
-        {
-            using (FileStream stream = new FileStream(path, FileMode.Open))
-            {
-                return await UploadAsync(
-                    stream,
-                    accessConditions: default, // Pass anything else so we don't recurse on this overload
-                    cancellationToken: cancellationToken)
-                    .ConfigureAwait(false);
-            }
-        }
+            CancellationToken cancellationToken) =>
+            await UploadAsync(
+                path,
+                overwrite: false,
+                cancellationToken: cancellationToken)
+                .ConfigureAwait(false);
 #pragma warning restore AZC0002 // Client method should have cancellationToken as the last optional parameter
 
         /// <summary>
-        /// The <see cref="Upload(Stream, BlobHttpHeaders, Metadata, BlobAccessConditions?, IProgress{long}, AccessTier?, StorageTransferOptions, CancellationToken)"/>
+        /// The <see cref="Upload(Stream, CancellationToken)"/> operation
+        /// creates a new block blob or updates the content of an existing
+        /// block blob.  Updating an existing block blob overwrites any
+        /// existing metadata on the blob.
+        ///
+        /// For partial block blob updates and other advanced features, please
+        /// see <see cref="BlockBlobClient"/>.  To create or modify page or
+        /// append blobs, please see <see cref="PageBlobClient"/> or
+        /// <see cref="AppendBlobClient"/>.
+        ///
+        /// For more information, see <see href="https://docs.microsoft.com/rest/api/storageservices/put-blob" />.
+        /// </summary>
+        /// <param name="content">
+        /// A <see cref="Stream"/> containing the content to upload.
+        /// </param>
+        /// <param name="overwrite">
+        /// Whether the upload should overwrite any existing blobs.  The
+        /// default value is false.
+        /// </param>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <returns>
+        /// A <see cref="Response{BlobContentInfo}"/> describing the
+        /// state of the updated block blob.
+        /// </returns>
+        /// <remarks>
+        /// A <see cref="RequestFailedException"/> will be thrown if
+        /// a failure occurs.
+        /// </remarks>
+        [ForwardsClientCalls]
+        public virtual Response<BlobContentInfo> Upload(
+            Stream content,
+            bool overwrite = false,
+            CancellationToken cancellationToken = default) =>
+            Upload(
+                content,
+                conditions: overwrite ? null : new BlobRequestConditions { IfNoneMatch = new ETag(Constants.Wildcard) },
```

Comment:
The wildcard makes the semantics here "update if the blob (however identified from its unique identifier) is not present."   
This means it will fail if you try to upload a blob that is the same version as the one you're holding, which technically is not overwriting it.  I would have expected IfNoneMatch to take the Etag of the blob, so it would fail if you're trying the update with a different version, which would overwrite existing data.  If theses are the semantics the service defines, though, I'm not concerned.

Reply:
Yes, 99% of the time this API is called we won't have an ETag to use.  That's part of why we called this `overwrite` instead of `onlyIfUnchanged` like you're doing in AppConfig since it's not relative to an existing ETag.  Any customer who wants that behavior could the overload that allows them to pass `BlobAccessConditions`.